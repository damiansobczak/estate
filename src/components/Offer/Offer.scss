@import "../../assets/styles/utils/variables";
@import "../../assets/styles/utils/mixins";

.offer {
  background-color: #fff;
  box-shadow: 0 40px 70px rgba(0, 0, 0, 0.08);
  border-radius: 4px;
  overflow: hidden;
  cursor: pointer;
  position: relative;
  top: 0;
  transition: all 0.2s ease-in-out;

  &:hover {
    box-shadow: 0 60px 70px rgba(0, 0, 0, 0.12);
    top: -2px;
    opacity: 0.7;
  }

  &__title {
    color: $color-text-primary;
  }

  &__media {
    position: relative;
    overflow: hidden;
    height: 245px;

    img {
      width: 100%;
      height: 100%;
      object-fit: cover;
      transition: transform 0.3s ease-in-out;
    }
  }

  &__price {
    padding: 8px 15px;
    background-color: rgba($color-accent, 0.8);
    position: absolute;
    top: 50px;
    right: 0;
    color: #fff;
    font-family: poppins, sans-serif;
    font-weight: 600;
    font-style: normal;
    font-size: $font-200;
  }

  &__content {
    padding: 25px;
  }

  &__meta {
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding-top: 10px;
  }

  &__category {
    font-weight: 700;
    color: $color-accent;
    text-transform: uppercase;
    font-size: $font-200;
    position: relative;
    padding-left: 15px;
    letter-spacing: 1px;

    &:before {
      content: "";
      position: absolute;
      left: 0;
      top: 50%;
      transform: translateY(-50%);
      height: 7px;
      width: 7px;
      background-color: $color-accent;
      border-radius: 50%;
    }
  }

  &__date {
    text-transform: uppercase;
    color: $color-text-secondary;
    font-size: $font-200;
  }

  &__title {
    font-family: poppins, sans-serif;
    font-weight: 600;
    font-style: normal;
    line-height: 1.2;
    margin: 25px 0;
  }

  &__sep {
    width: 50px;
    height: 1px;
    background-color: rgba($color-neutral-300, 0.5);
    margin-bottom: 25px;
  }

  &__tags {
    color: $color-text-secondary;
    font-size: $font-200;
    font-weight: 600;
  }
}
