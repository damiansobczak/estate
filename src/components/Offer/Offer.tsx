import React from "react";
import "./Offer.scss";
import { OfferProps } from "./Interfaces/OfferProps";

class Offer extends React.Component<OfferProps, any> {
  render() {
    return (
      <div className="column is-6-tablet is-4-desktop">
        <div className="offer">
          <div className="offer__media">
            <img src={this.props.image} alt="" />
            <div className="offer__favorite" />
            <div className="offer__price">{this.props.price}</div>
          </div>
          <div className="offer__content">
            <div className="offer__meta">
              <div className="offer__category">{this.props.category}</div>
              <div className="offer__date">{this.props.date}</div>
            </div>
            <div className="offer__title">{this.props.title}</div>
            <div className="offer__sep" />
            <div className="offer__tags">{this.props.tags.map((tag: string) => `${tag}, `)}</div>
          </div>
        </div>
      </div>
    );
  }
}

export default Offer;
