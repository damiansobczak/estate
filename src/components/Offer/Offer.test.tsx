import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Offer from "./Offer";
import { BrowserRouter as Router } from 'react-router-dom';


describe("<Offer />", () => {
    it("renders component correctly", () => {
        const tree = renderer.create(<Router><Offer id={1} image="" price="145,000zł" category="Wynajem" date="01 Czer, 2020" title="Ostrów Wielkopolski" tags={["2 pokoje", "Łazienka", "Parking"]} /></Router>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it("displays image when empty props", () => {
        const wrapper = shallow(<Router><Offer id={1} image="" price="145,000zł" category="Wynajem" date="01 Czer, 2020" title="Ostrów Wielkopolski" tags={["2 pokoje", "Łazienka", "Parking"]} /></Router>);
        expect(wrapper.html()).toContain('<img src="image-1.jpg"');
    });

    it("displays tags in correct way", () => {
        const wrapper = shallow(<Router><Offer id={1} image="" price="145,000zł" category="Wynajem" date="01 Czer, 2020" title="Ostrów Wielkopolski" tags={["2 pokoje", "Łazienka", "Parking"]} /></Router>);
        expect(wrapper.html()).toContain('2 pokoje, Łazienka, Parking');
    });
});
