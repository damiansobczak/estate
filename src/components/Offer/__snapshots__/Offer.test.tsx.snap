// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Offer /> renders component correctly 1`] = `
<a
  className="column is-6-tablet is-4-desktop"
  href="/oferta/1"
  onClick={[Function]}
>
  <div
    className="offer__favorite "
    onClick={[Function]}
  >
    <span
      className="icon-heart"
    />
  </div>
  <div
    className="offer"
  >
    <div
      className="offer__media"
    >
      <img
        alt=""
        src="image-1.jpg"
      />
      <div
        className="offer__price"
      >
        145,000zł
      </div>
    </div>
    <div
      className="offer__content"
    >
      <div
        className="offer__meta"
      >
        <div
          className="offer__category"
        >
          Wynajem
        </div>
        <div
          className="offer__date"
        >
          01 Czer, 2020
        </div>
      </div>
      <div
        className="offer__title"
      >
        Ostrów Wielkopolski
      </div>
      <div
        className="offer__sep"
      />
      <div
        className="offer__tags"
      >
        2 pokoje, 
        Łazienka, 
        Parking, 
      </div>
    </div>
  </div>
</a>
`;
